src/Pointer.ts

```diff
@@ -31,16 +40,28 @@ export class Pointer {
     public draw(): void {
         if (this.destination === null) return;
 
-        const pointerAngle = this.origin.getAngleTo(this.destination);
+        let x1, y1, x2, y2, pointerAngle;
+        if (this.line instanceof fabric.Path) {
+            pointerAngle = 0.85;
+            ({ x: x1, y: y1 } = this.origin.getTailContactPoint(0));
+            ({ x: x2, y: y2 } = this.destination.getHeadContactPoint(0));
 
-        const { x: x1, y: y1 } = this.origin.getTailContactPoint(pointerAngle);
-        const { x: x2, y: y2 } = this.destination.getHeadContactPoint(pointerAngle);
-        const arrowAngles = [pointerAngle + 0.85 * Math.PI, pointerAngle - 0.85 * Math.PI];
+            const size = Config.NODE_SIZE * 3;
+            this.line.set("path", [
+                <any>["m", x1, y1],
+                <any>["c", size , -size , -size * 3/2, -size, x2 - x1, 0]
```

Comment:
@jasonxia17 I had to put this `<any>` here because the fabric.js typescript declarations don't actually match up with the implementation, but this causes the linter to throw some warnings, which I also don't like. 

Any ideas on how to handle that?

Reply:
I don't like the fact that `this.line` can oscillate between two different types. I'd rather have two distinct member variables:

```
private line: fabric.Line;
private selfLoop: fabric.Path;
```

Then, inside the `draw` function, we can call a `setVisibilities` function that make one of them visible and the other invisible. Or, if the destination is null, we should make both of them invisible, and also make the arrowheads invisible.

Reply:
Ok, I can do that. 

That's actually disjoint from the problem I was raising here, which is that the type declaration for `fabric.Path.path` is incorrect, requiring me to add a dirty type cast.

Reply:
I'm not too familiar with TypeScript, so I don't know how to handle this issue either.

Reply:
I think I'll just leave it for now. If the compile warning get too annoying, I can just turn off that particular warning in the compiler, since I don't think either of us are going to accidently put unneccesary casts to `<any>` into the codebase.

Reply:
I just figured out a way around it. I can cast to `<unknown>` and then to `<fabric.Point>` and that makes the compiler happy. 

I'll leave a comment about it too.